README.md
Magic the Gathering Planechase Utility
======================================

For use with Magic the Gathering [Planechase](https://mtg.gamepedia.com/Planechase) games

Features
--------

- Single-deck Planechase featuring all published planes
- Automatic planeswalking when virtual die roller achieves planeswalk
- [Eternities Map](https://web.archive.org/web/20220922215232/https://magic.wizards.com/en/articles/\
archive/feature/eternities-map-2010-07-19-0) variant
  - Phenomena occur only during Hellrides.

Shortcuts
------------------

Key | Function | Mouse | On
--- | -------- | ----- | --
`<SPACE>` | Reset planar die mana cost | Click on Cost number | Planechase / Eternties Map
`<ENTER>` `<R>` | Roll planar die | Click on Roll button | Planechase / Eternties Map
`<C>` `<=>` `<+>` | Add counter to current plane | Click on current plane | Planechase / Eternities Map
`<ESC>` | Toggle card-specific view | Click on 'Toggle' link in toolbar | Planechase / Eternities Map

The Eternities Map can be navigated by use of arrow keys, vi navigation (HJKL), and cardinal direction keys (NSEW).

Options
-------

- URL Parameters
  - `bigpreview=1` : enable large previews of planes when choosing which cards to include.
  - `chroma=1` : set background to chromakey green.
  - `size=2` : change the radius of visible/recycled planes on the eternities map. Default radius is 3, maximum 3.
  - `manual=1` : enable manual mode: disable automatic planeswalking. Right arrow manually planeswalks.
  - `custom=1` : enable custom planes from [Multiverse](https://www.reddit.com/r/custommagic/comments/7sgecl/multiverse_the_custom_planechase_expansion/), [digitalmayhemx](https://www.reddit.com/u/digitalmayhemx) and other potential sources

Norn's Seedcore
---------------
Norn's Seedcore is a new plane from the March of the Machine Commander
decks. It reads:

> When you planeswalk to Norn’s Seedcore, chaos ensues.
>
> Whenever chaos ensues, reveal cards from the top of your planar deck
> until you reveal a plane card. Planeswalk to it, except don’t
> planeswalk away from any plane. Put the rest of the revealed cards
> on the bottom of vour planar deck in any order.

One-at-a-time Planechase will reveal an additional plane with Norn's
Seedcore, and subsequent Chaos rolls will add additional planes to the
display. Planeswalking away from Norn's Seedcore will shuffle all
previously-associated planes with it into the Planes deck.

When planeswalking to Norn's Seedcore on the Eternities Map, the
interface will automatically reveal an additional plane. To return to
the map, click on Norn's Seedcore. If you return to Norn's Seedcore
while it is still on the map, any planes previously associated with it
will still be linked to it, and they will only be removed once Norn's
Seedcore has been removed from the map.

If you encounter a plane with plane-specific counters, pressing the
counter button in the toolbar will reveal all pertinent counter
values.

Note: A plane may both be on the map and associated with Norn's
Seedcore simultaneously.

Adding Custom Planes
--------------------

To add custom planes, update the `eternity` object located at the top
of `planechase.js`. Add the details of each plane under the
`custom_planes` key--each value should be an object with a `name`
(preferably with an indicator that it is different from WOTC-created
planes) and `type` where type can be either `plane` or `phenomenon`.

Submit a pull request with those changes, and I'll double-check them 
and update the site accordingly. 

TODO
----

* BUG: Sometimes Spatial Merging will show up in the map unbidden (EM)
* BUG: Sometimes the phenomenon/chaos divs will get into a bad toggle state
  (e.g. hidden when the plane div is hidden).
* BUG: Resetting the mana cost with the spacebar sometimes kicks off a roll of the planar die.
* BUG: Spatial Merging should allow for counter-enabled planes
* TODO: Pools of Becoming + Stairs to Infinity
* TODO: change get_next_planes to go through deck until plane
* TODO: full manual mode (no processing, enable scrying, game history);
* TODO: implement Fixed Point in Time

Contact
-------

Please direct any questions, feature requests, etc. to the github repository or to [@thepeopleseason](https://twitter.com/thepeopleseason) on Twitter.


Changelog.md
# Changelog

## 2023-10-06

### Added

- Added MTGWHO Planes from today's previews.

## 2023-10-05

### Added

- Added MTGWHO Planes from today's previews.

## 2023-10-04

### Added

- Added MTGWHO Planes from today's previews.

## 2023-10-03

### Added

- Added MTGWHO Planes from today's previews.

## 2023-10-02

### Added

- Added MTGWHO Planes from today's previews.

## 2023-07-28

### Added

- Added MTGWHO Planes from current previews.

## 2023-05-24

### Changed

- Changed options from urlParams to checkboxes

## 2023-05-22

### Changed

- Changed structure of the main eternity object to simplify process of adding custom planes.

## 2023-05-19

### Added

- Added [digitalmayhemx](https://imgur.com/a/hf4TP) planes

### Changed

- `mv=1` is now `custom=1` URL parameter.

## 2023-05-16

### Added

- Added [Multiverse](https://www.reddit.com/r/custommagic/comments/7sgecl/multiverse_the_custom_planechase_expansion/) planes and `mv=1` URL parameter.

## 2023-04-30

### Added

- Enabled WakeLocking for devices (hopefully)

## 2023-04-29

### Added

- Changelog

## 2023-04-28

### Added

- `chroma=1` URL paramter will enable green background for chromakeying.
- `bigpreview=1` URL parameter enables large previews on card chooser screen.
- [`blankandblank/`](https://thepeopleseason.github.io/planechase/blankandblank/) for random generation of a [Sword of _ and _](https://www.tcgplayer.com/product/485192/magic-un-known-event-playtest-cards-sword-of-_-and-_?Language=English)

### Changed

- Eternities Map plane size determined by width of screen, to better fit full map onto screen.

## 2023-04-24

### Added

- Enabled choosing of cards to include in games
- Copyright Footer and Help Messages
- localStorage saves most recently chosen cards

### Fixed

- Starting on Norn's Seedcore no longer breaks the Eternities Map

## 2023-04-22

### Added

- Toggle links to view alternate `div` from primary plane/eternities map div.

### Fixed

- Properly handles Norn's Seedcore when flipped to from a chaos roll from Pools of Becoming.

## 2023-04-20

### Added

- Planar counter memory and processing.
- Documentation for Norn's Seedcore

## 2023-04-18

### Added

- Handle Norn's Seedcore when chaos ensues.
- Manual mode -- do not planeswalk when included dice roller hits Planeswalk.

### Fixed

- Stairs to Infinity now allows users to keep revealed card on top of the planar deck
- Don't reset costs when Planeswalking

## 2023-04-12

### Added

- New planes from MOC

## 2023-03-31

### Added

- Preview planes from MOC

## 2023-01-15

### Added

- LICENSE
- `size=x` parameter to control size of Eternities Map
- Expanded keyboard shortcuts

## 2020-05-06

### Fixed

- UI, Chaotic Aether, Hellriding

## 2020-05-01

### Fixed

- Planar counter tracking
- Spatial Merging processing

## 2020-04-29

### Added

- Phenomena in the Eternities Map -- happens only when Hellriding

## 2020-04-28

### Added

- Keyboard shortcuts

## 2014-11-14

### Added

- Initial release of Planechase


archenemy/rename_scryfall_files.sh
#!/bin/sh
#
# Rename files from scryfall to remove unicode filename

mv oe01-9★-make-yourself-useful.png oe01-9-make-yourself-useful.png
mv oe01-8★-know-evil.png oe01-8-know-evil.png
mv oe01-7★-for-each-of-you-a-gift.png oe01-7-for-each-of-you-a-gift.png
mv oe01-6★-every-dream-a-nightmare.png oe01-6-every-dream-a-nightmare.png
mv oe01-5★-delight-in-the-hunt.png oe01-5-delight-in-the-hunt.png
mv oe01-4★-choose-your-demise.png oe01-4-choose-your-demise.png
mv oe01-3★-bow-to-my-command.png oe01-3-bow-to-my-command.png
mv oe01-2★-behold-my-grandeur.png oe01-2-behold-my-grandeur.png
mv oe01-20★-when-will-you-learn.png oe01-20-when-will-you-learn.png
mv oe01-1★-because-i-have-willed-it.png oe01-1-because-i-have-willed-it.png
mv oe01-19★-what-s-yours-is-now-mine.png oe01-19-what-s-yours-is-now-mine.png
mv oe01-18★-this-world-belongs-to-me.png oe01-18-this-world-belongs-to-me.png
mv oe01-17★-there-is-no-refuge.png oe01-17-there-is-no-refuge.png
mv oe01-16★-a-reckoning-approaches.png oe01-16-a-reckoning-approaches.png
mv oe01-15★-power-without-equal.png oe01-15-power-without-equal.png
mv oe01-14★-pay-tribute-to-me.png oe01-14-pay-tribute-to-me.png
mv oe01-13★-no-one-will-hear-your-cries.png oe01-13-no-one-will-hear-your-cries.png
mv oe01-12★-my-laughter-echoes.png oe01-12-my-laughter-echoes.png
mv oe01-11★-my-forces-are-innumerable.png oe01-11-my-forces-are-innumerable.png
mv oe01-10★-the-mighty-will-fall.png oe01-10-the-mighty-will-fall.png
mv oarc-9★-embrace-my-diabolical-vision.png oarc-9-embrace-my-diabolical-vision.png
mv oarc-8★-a-display-of-my-dark-power.png oarc-8-a-display-of-my-dark-power.png
mv oarc-7★-the-dead-shall-serve.png oarc-7-the-dead-shall-serve.png
mv oarc-6★-dance-pathetic-marionette.png oarc-6-dance-pathetic-marionette.png
mv oarc-5★-choose-your-champion.png oarc-5-choose-your-champion.png
mv oarc-4★-behold-the-power-of-destruction.png oarc-4-behold-the-power-of-destruction.png
mv oarc-45★-your-will-is-not-your-own.png oarc-45-your-will-is-not-your-own.png
mv oarc-44★-your-puny-minds-cannot-fathom.png oarc-44-your-puny-minds-cannot-fathom.png
mv oarc-43★-your-fate-is-thrice-sealed.png oarc-43-your-fate-is-thrice-sealed.png
mv oarc-42★-which-of-you-burns-brightest.png oarc-42-which-of-you-burns-brightest.png
mv oarc-41★-the-very-soil-shall-shake.png oarc-41-the-very-soil-shall-shake.png
mv oarc-40★-tooth-claw-and-tail.png oarc-40-tooth-claw-and-tail.png
mv oarc-3★-approach-my-molten-realm.png oarc-3-approach-my-molten-realm.png
mv oarc-39★-surrender-your-thoughts.png oarc-39-surrender-your-thoughts.png
mv oarc-38★-rotted-ones-lay-siege.png oarc-38-rotted-ones-lay-siege.png
mv oarc-37★-roots-of-all-evil.png oarc-37-roots-of-all-evil.png
mv oarc-36★-realms-befitting-my-majesty.png oarc-36-realms-befitting-my-majesty.png
mv oarc-35★-the-pieces-are-coming-together.png oarc-35-the-pieces-are-coming-together.png
mv oarc-34★-only-blood-ends-your-nightmares.png oarc-34-only-blood-ends-your-nightmares.png
mv oarc-33★-nothing-can-stop-me-now.png oarc-33-nothing-can-stop-me-now.png
mv oarc-32★-nature-shields-its-own.png oarc-32-nature-shields-its-own.png
mv oarc-31★-nature-demands-an-offering.png oarc-31-nature-demands-an-offering.png
mv oarc-30★-my-wish-is-your-command.png oarc-30-my-wish-is-your-command.png
mv oarc-2★-all-shall-smolder-in-my-wake.png oarc-2-all-shall-smolder-in-my-wake.png
mv oarc-29★-my-undead-horde-awakens.png oarc-29-my-undead-horde-awakens.png
mv oarc-28★-my-genius-knows-no-bounds.png oarc-28-my-genius-knows-no-bounds.png
mv oarc-27★-my-crushing-masterstroke.png oarc-27-my-crushing-masterstroke.png
mv oarc-26★-mortal-flesh-is-weak.png oarc-26-mortal-flesh-is-weak.png
mv oarc-25★-may-civilization-collapse.png oarc-25-may-civilization-collapse.png
mv oarc-24★-look-skyward-and-despair.png oarc-24-look-skyward-and-despair.png
mv oarc-23★-know-naught-but-fire.png oarc-23-know-naught-but-fire.png
mv oarc-22★-the-iron-guardian-stirs.png oarc-22-the-iron-guardian-stirs.png
mv oarc-21★-introductions-are-in-order.png oarc-21-introductions-are-in-order.png
mv oarc-20★-into-the-earthen-maw.png oarc-20-into-the-earthen-maw.png
mv oarc-1★-all-in-good-time.png oarc-1-all-in-good-time.png
mv oarc-19★-ignite-the-cloneforge.png oarc-19-ignite-the-cloneforge.png
mv oarc-18★-i-know-all-i-see-all.png oarc-18-i-know-all-i-see-all.png
mv oarc-17★-i-delight-in-your-convulsions.png oarc-17-i-delight-in-your-convulsions.png
mv oarc-16★-i-call-on-the-ancient-magics.png oarc-16-i-call-on-the-ancient-magics.png
mv oarc-15★-i-bask-in-your-silent-awe.png oarc-15-i-bask-in-your-silent-awe.png
mv oarc-14★-feed-the-machine.png oarc-14-feed-the-machine.png
mv oarc-13★-the-fate-of-the-flammable.png oarc-13-the-fate-of-the-flammable.png
mv oarc-12★-evil-comes-to-fruition.png oarc-12-evil-comes-to-fruition.png
mv oarc-11★-every-last-vestige-shall-rot.png oarc-11-every-last-vestige-shall-rot.png
mv oarc-10★-every-hope-shall-vanish.png oarc-10-every-hope-shall-vanish.png
